nifi-extension-bundles/nifi-aws-bundle/nifi-aws-kinesis/src/main/java/org/apache/nifi/processors/aws/kinesis/ConsumeKinesis.java

```diff
@@ -0,0 +1,652 @@
+        The maximum size of the buffer is controlled by the 'Max Bytes to Buffer' property.
+        In addition, Kinesis Client Library may cache some amount of data for each shard when the processor's buffer is full.""")
+public class ConsumeKinesis extends AbstractProcessor {
+
+    private static final Duration HTTP_CLIENTS_CONNECTION_TIMEOUT = Duration.ofSeconds(30);
+    private static final Duration HTTP_CLIENTS_READ_TIMEOUT = Duration.ofMinutes(3);
+
+    /**
+     * Best balance between throughput and CPU usage by KCL.
+     */
+    private static final int KINESIS_HTTP_CLIENT_CONCURRENCY_PER_TASK = 16;
+    private static final int KINESIS_HTTP_CLIENT_WINDOW_SIZE_BYTES = 512 * 1024; // 512 KiB
+    private static final Duration KINESIS_HTTP_HEALTH_CHECK_PERIOD = Duration.ofMinutes(1);
+
+    static final PropertyDescriptor KINESIS_STREAM_NAME = new PropertyDescriptor.Builder()
+            .name("Amazon Kinesis Stream Name")
+            .description("The name of the Kinesis stream to consume from.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor APPLICATION_NAME = new PropertyDescriptor.Builder()
+            .name("Application Name")
+            .description("The name of the Kinesis application. This is used for DynamoDB table naming and worker coordination.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor AWS_CREDENTIALS_PROVIDER_SERVICE = new PropertyDescriptor.Builder()
+            .name("AWS Credentials Provider Service")
+            .description("The Controller Service that is used to obtain AWS credentials provider.")
+            .required(true)
+            .identifiesControllerService(AWSCredentialsProviderService.class)
+            .build();
+
+    static final PropertyDescriptor REGION = new PropertyDescriptor.Builder()
+            .name("Region")
+            .description("AWS Region in which the Kinesis stream is located.")
+            .required(true)
+            .allowableValues(RegionUtilV2.getAvailableRegions())
+            .defaultValue(RegionUtilV2.createAllowableValue(Region.US_WEST_2).getValue())
+            .build();
+
+    static final PropertyDescriptor RECORD_READER = new PropertyDescriptor.Builder()
+            .name("Record Reader")
+            .description("""
+                    The Record Reader to use for parsing the data received from Kinesis.
+                    If not set, the records are written one per FlowFile without parsing.
+
+                    The Record Reader is responsible for providing schemas for the records. If the schemas change frequently,
+                    it might hinder performance of the processor. (See processor's additional details for more information.)""")
+            .required(false)
```

Comment:
Instead making this optional, recommend adding a new `Processing Strategy` property with `FLOW_FILE` and `RECORD` as values, then making these and Record Writer depend on the Record strategy.